muffnn/mlp/mlp_classifier.py

```diff
@@ -111,6 +111,9 @@ def _init_model_output(self, t):
                 t = tf.nn.dropout(t, keep_prob=self._keep_prob)
             t = affine(t, output_size, scope='output_layer')
 
+        # TODO: discuss where to move the instantiaion of self.input_targets_
+        # TODO: it need to be moved or the init_model_output needs to take an addition input
+
```

Comment:
To preserve the API of the internal objects, we need to leave this here. IDK if it is a problem but I think it should be ok.

Reply:
I'll try to think of a work around, but `self.input_targets_` needs to be passed into the dataset API. I'm thinking that maybe we can zip up the datasets after the fact. However I'm not exactly sure how the synchronization would work in such a case.

Reply:
hmmmmm. I need to see the code more to understand this. A close to working version with just this issue would help me.